capstone-frontend/src/components/CreateClub.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
+import '../styles/Groups.css';
+
+class CreateClub extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(e) {
+    const history = this.props.history;
+    e.preventDefault();
+    let data = {};
+    const formElements = document.getElementById("create-club-form").elements;
+    for (let i = 0; i < formElements.length; i++) {
+      if (formElements[i].type !== "submit") {
+        data[formElements[i].name] = formElements[i].value;
+      }
+    }
+    fetch("/api/clubs", 
+        {method: "post", body: JSON.stringify(data)})
```

Comment:
Love the pattern you're using here to handle post requests - i.e., adding JSON key/value pairs to the request body and then reading data in the request's body in `createRequestBodyJson.` It really keeps the code concise, even as the size of your `form` grows.

Reply:
Thanks! I considered using the jquery serialize function instead, but it turns the form fields into a query string (eg. ?id=<id> etc) instead of a JSON body. Decided to do this instead, and make it a loop to keep it general so it can be extended as forms grow.